Mathlib/Analysis/SpecialFunctions/Pow/NNReal.lean

```diff
@@ -812,6 +824,8 @@ theorem rpow_pos_of_nonneg {p : ℝ} {x : ℝ≥0∞} (hx_pos : 0 < x) (hp_nonne
     rw [← zero_rpow_of_pos hp_pos]
     exact rpow_lt_rpow hx_pos hp_pos
 
+-- This is an unsafe rule since we want to try `rpow_ne_top_of_nonneg` if x = 0.
+@[aesop (rule_sets := [finiteness]) unsafe apply]
```

Comment:
Why is that a finiteness rule? It doesn't conclude finiteness.

Reply:
Good point! I meant to state a different lemma, added now.

Reply:
Added a test that this works.